api/src/operations/exec/index.test.ts

```diff
@@ -1,21 +1,76 @@
-import { VMError } from 'vm2';
 import { test, expect } from 'vitest';
-
 import config from './index.js';
 
-test('Rejects when modules are used without modules being allowed', async () => {
+test('Rejects when Isolate uses more than allowed memory', async () => {
+	const testCode = `
+		const storage = [];
+		const twoMegabytes = 1024 * 1024 * 2;
+		while (true) {
+			const array = new Uint8Array(twoMegabytes);
+			for (let ii = 0; ii < twoMegabytes; ii += 4096) {
+				array[ii] = 1; // we have to put something in the array to flush to real memory
+			}
+			storage.push(array);
+		}
+	`;
+
+	await expect(
+		config.handler({ code: testCode }, {
+			data: {},
+			env: {
+				FLOWS_RUN_SCRIPT_MAX_MEMORY: 8,
+				FLOWS_RUN_SCRIPT_TIMEOUT: 10000,
+			},
+		} as any)
+	).rejects.toThrow('Array buffer allocation failed');
+});
+
+test('Rejects when operation runs for longer than allowed ', async () => {
+	const testCode = `
+		while (true) {}
+	`;
```

Comment:
This `testCode` covers the following timeout.

https://github.com/directus/directus/blob/b87c60829527e18c2c3a90bd2780bbbfc39c12f6/api/src/operations/exec/index.ts#L47

But the following timeout isn't tested.

https://github.com/directus/directus/blob/b87c60829527e18c2c3a90bd2780bbbfc39c12f6/api/src/operations/exec/index.ts#L51-L54

I think we should add an additional test to cover long running functions with the following `testCode`.

```js
	const testCode = `
		module.exports = async function (data) {
			while (true) {}
		};
	`;
```

Reply:
Thanks - Done!